src/Runner.Worker/Container/ContainerHooks/ContainerHookManager.cs

```diff
@@ -0,0 +1,215 @@
+                ResponseFile = responsePath,
+                Args = new ScriptStepArgs
+                {
+                    Container = container.GetHookContainer(),
+                    EntryPointArgs = entryPointArgs.Split(' ').Select(arg => arg.Trim()),
+                    EntryPoint = entryPoint,
+                    EnvironmentVariables = environmentVariables,
+                    PrependPath = prependPath,
+                    WorkingDirectory = workingDirectory,
+                },
+                State = GetHookStateInJson(context)
+            };
+
+            var response = await ExecuteHookScript(context, input, ActionRunStage.Main);
+            SaveHookState(context, response.State);
+            
+        }
+
+        private async Task<HookResponse> ExecuteHookScript(IExecutionContext context, HookInput input, ActionRunStage stage)
+        {
+            var scriptDirectory = Path.GetDirectoryName(HookIndexPath);
+            var stepHost = HostContext.CreateService<IDefaultStepHost>();
+            Dictionary<string, string> inputs = new()
+            {
+                ["standardInInput"] = JsonUtility.ToString(input),
+                ["path"] = HookIndexPath,
+                ["shell"] = Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Externals), NodeUtil.GetInternalNodeVersion(), "bin", $"node{IOUtil.ExeExtension}") + " {0}" // TODO: fix hardcoded node path
+            };
+
+            var handlerFactory = HostContext.GetService<IHandlerFactory>();
+            var handler = handlerFactory.Create(
+                            context,
+                            null,
+                            stepHost,
+                            new ScriptActionExecutionData(),
+                            inputs,
+                            environment: new Dictionary<string, string>(VarUtil.EnvironmentVariableKeyComparer),
+                            context.Global.Variables,
+                            actionDirectory: scriptDirectory,
+                            localActionContainerSetupSteps: null) as ScriptHandler;
+            handler.PrepareExecution(stage); // TODO: find out stage, we only use Start in pre, but double check
+
+            IOUtil.CreateEmptyFile(input.ResponseFile);
+            await handler.RunAsync(stage);
+            if (handler.ExecutionContext.Result == TaskResult.Failed)
+            {
+                throw new Exception("Hook failed"); // TODO: better exception
+            }
+
+            // TODO: consider the case when the responseFile doesn't exist - should we throw or noop on response?
+            var response = IOUtil.LoadObject<HookResponse>(input.ResponseFile);
+            IOUtil.DeleteFile(input.ResponseFile);
+            return response;
+        }
+
+        private string GenerateResponsePath()
+        {
+            return $"{Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Temp), ResponseFolderName)}/{Guid.NewGuid()}.json";
```

Comment:
Can we use Path.Combine for whole path? Something like this:
Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Temp), ResponseFolderName, $"{Guid.NewGuid()}.json")